docs/sql-ref-functions-builtin-aggregate.md

```diff
@@ -19,4 +19,657 @@ license: |
   limitations under the License.
 ---
 
-Aggregate functions
\ No newline at end of file
+Spark SQL provides build-in aggregate functions defined in the dataset API and SQL interface. Aggregate functions
+operate on a group of rows and return a single aggregated value.
+
+<table class="table">
+  <thead>
+    <tr><th style="width:25%">Function</th><th>Argument Type(s)</th><th>Description</th></tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td><b>{any | some | bool_or}</b>(<i>expression</i>)</td>
+      <td>boolean</td>
+      <td>Returns true if at least one value is true.</td>
+    </tr>
+    <tr>
+      <td><b>approx_count_distinct</b>(<i>expression[, relativeSD]</i>)</td>
+      <td>(long, double)</td>
+      <td>`relativeSD` is the maximum estimation error allowed. Returns the estimated cardinality by HyperLogLog++.</td>
+    </tr>   
+    <tr>
+      <td><b>{avg | mean}</b>(<i>expression</i>)</td>
+      <td>short, float, byte, decimal, double, int, long or string</td>
+      <td>Returns the average of values in the input expression.</td> 
+    </tr>
+    <tr>
+      <td><b>{bool_and | every}</b>(<i>expression</i>)</td>
+      <td>boolean</td>
+      <td>Returns true if all values are true.</td>
+    </tr>
+    <tr>
+      <td><b>collect_list</b>(<i>expression</i>)</td>
+      <td>any</td>
+      <td>Collects and returns a list of non-unique elements. The function is non-deterministic because the order of collected results depends on the order of the rows which may be non-deterministic after a shuffle.</td>
+    </tr>       
+    <tr>
+      <td><b>collect_set</b>(<i>expression</i>)</td>
+      <td>any</td>
+      <td>Collects and returns a set of unique elements. The function is non-deterministic because the order of collected results depends on the order of the rows which may be non-deterministic after a shuffle.</td>
+    </tr>
+    <tr>
+      <td><b>corr</b>(<i>expression1, expression2</i>)</td>
+      <td>(double, double)</td>
+      <td>Returns Pearson coefficient of correlation between a set of number pairs.</td>
+    </tr>
+    <tr>
+      <td><b>count</b>([<b>DISTINCT</b>] <i>*</i>)</td>
+      <td>none</td>
+      <td>If specified <code>DISTINCT</code>, returns the total number of retrieved rows are unique and not null; Otherwise, returns the total number of retrieved rows, including rows containing null.</td>
+    </tr>
+    <tr>
+      <td><b>count</b>([<b>DISTINCT</b>] <i>expression1[, expression2</i>])</td>
+      <td>(any, any)</td>
+      <td>If specified <code>DISTINCT</code>, returns the number of rows for which the supplied expression(s) are unique and not null; Otherwise, returns the number of rows for which the supplied expression(s) are all not null.</td>
+    </tr>
+    <tr>
+      <td><b>count_if</b>(<i>predicate</i>)</td>
+      <td>expression that will be used for aggregation calculation</td>
```

Comment:
How about `expression that will be used for aggregation calculation` -> `expression that returns a boolean value`?

Reply:
yah, this is better. Done